src/App.test.js

```diff
@@ -653,12 +666,104 @@ describe('App component', () => {
+            >
+              <App locale="en" />
+            </Router>
+          </AppServicesProvider>,
+        );
+      });
+
+      // Assert
+      const localStorageItems = dependencies.localStorage.getAllItems();
+      await waitFor(() => {
+        expect(localStorageItems['UtmCookies']).toBeDefined();
+        expect(localStorageItems['UtmCookies']).toMatch('UTMSource');
+        expect(localStorageItems['UtmCookies']).toMatch('test');
+        expect(localStorageItems['UtmCookies']).toMatch('UTMCampaign');
+        expect(localStorageItems['UtmCookies']).toMatch('testcampaign');
+        expect(localStorageItems['UtmCookies']).toMatch('UTMMedium');
+        expect(localStorageItems['UtmCookies']).toMatch('testmedium');
+        expect(localStorageItems['UtmCookies']).toMatch('UTMTerm');
+        expect(localStorageItems['UtmCookies']).toMatch('testterm');
+      });
+    });
+
+    it('check utm parameters', async () => {
+      // Arrange
+      const dependencies = {
+        sessionManager: createDoubleSessionManager(),
+        localStorage: createLocalStorageDouble(),
+        dopplerSitesClient: dopplerSitesClientDouble,
+      };
+
+      const utmCookie = {
+        date: new Date().toISOString(),
+        UTMSource: 'utmsource1',
+        UTMCampaign: 'utmcampaign1',
+        UTMMedium: 'utmmedium1',
+        UTMTerm: 'utmterm1',
+      };
+      dependencies.localStorage.setItem('UtmCookies', JSON.stringify(utmCookie));
+      createJsonParse(utmCookie);
+      // Act
+      act(() => {
+        render(
+          <AppServicesProvider forcedServices={dependencies}>
+            <Router
+              initialEntries={[
+                '/signup?origin=testOrigin&utm_source=test&utm_campaign=testcampaign&utm_medium=testmedium&utm_term=testterm',
+              ]}
+            >
+              <App locale="en" />
+            </Router>
+          </AppServicesProvider>,
+        );
+      });
+
+      // Assert
+      const localStorageItems = dependencies.localStorage.getAllItems();
+      await waitFor(() => {
+        expect(localStorageItems['UtmCookies']).toBeDefined();
+        expect(localStorageItems['UtmCookies']).toMatch('UTMSource');
+        expect(localStorageItems['UtmCookies']).toMatch('test');
+        expect(localStorageItems['UtmCookies']).toMatch('UTMCampaign');
+        expect(localStorageItems['UtmCookies']).toMatch('testcampaign');
+        expect(localStorageItems['UtmCookies']).toMatch('UTMMedium');
+        expect(localStorageItems['UtmCookies']).toMatch('testmedium');
+        expect(localStorageItems['UtmCookies']).toMatch('UTMTerm');
+        expect(localStorageItems['UtmCookies']).toMatch('testterm');
+        expect(localStorageItems['UtmCookies']).toMatch('utmsource1');
+        expect(localStorageItems['UtmCookies']).toMatch('utmcampaign1');
+        expect(localStorageItems['UtmCookies']).toMatch('utmmedium1');
+        expect(localStorageItems['UtmCookies']).toMatch('utmterm1');
```

Comment:
Could you deserialize UtmCookies and count two instances (two items)?

Reply:
i try but its a string and when try to use json.parse in the `act` doesnt work, if you have an idea i can try